test-clients/src/main/java/com/hedera/services/bdd/suites/reconnect/ValidateTokensStateAfterReconnect.java

```diff
@@ -0,0 +1,214 @@
+		String adminKey = "adminKey";
+		String newAdminKey = "newAdminKey";
+
+		return customHapiSpec("ValidateTokensAfterReconnect")
+				.withProperties(Map.of(
+						"txn.start.offset.secs", "-5")
+				)
+				.given(
+						sleepFor(Duration.ofSeconds(25).toMillis()),
+						tokenOpsEnablement(),
+						newKeyNamed(supplyKey),
+						newKeyNamed(freezeKey),
+						newKeyNamed(adminKey),
+						newKeyNamed(newAdminKey),
+						cryptoCreate(TOKEN_TREASURY).balance(ONE_MILLION_HBARS).logging(),
+						cryptoCreate(anotherAccount).balance(ONE_HUNDRED_HBARS).logging()
+				)
+				.when(
+						getAccountBalance(GENESIS)
+								.setNode(reconnectingNode)
+								.unavailableNode(),
+
+						tokenCreate(tokenToBeQueried)
+								.freezeKey(freezeKey)
+								.supplyKey(supplyKey)
+								.initialSupply(TOKEN_INITIAL_SUPPLY)
+								.treasury(TOKEN_TREASURY)
+								.adminKey(adminKey)
+								.logging(),
+						tokenCreate(anotherToken)
+								.freezeKey(freezeKey)
+								.supplyKey(supplyKey)
+								.initialSupply(TOKEN_INITIAL_SUPPLY)
+								.adminKey(adminKey)
+								.treasury(TOKEN_TREASURY)
+								.logging(),
+
+						/* Some token operations*/
+						getTokenInfo(tokenToBeQueried).logged(),
+						getTokenInfo(anotherToken).logged(),
+
+						tokenUpdate(tokenToBeQueried)
+								.fee(ONE_HUNDRED_HBARS)
+								.payingWith(TOKEN_TREASURY)
+								.adminKey(newAdminKey).logging(),
+
+						tokenAssociate(anotherAccount, tokenToBeQueried, anotherToken).logging(),
+						blockingOrder(
+								IntStream.range(0, 10).mapToObj(i ->
+										cryptoTransfer(moving(1, tokenToBeQueried)
+												.between(TOKEN_TREASURY, anotherAccount))
+												.logging())
+										.toArray(HapiSpecOperation[]::new)
+						),
+						blockingOrder(
+								IntStream.range(0, 5).mapToObj(i ->
+										mintToken(tokenToBeQueried, 100).logging())
+										.toArray(HapiSpecOperation[]::new)
+						),
+						blockingOrder(
+								IntStream.range(0, 5).mapToObj(i ->
+										mintToken(anotherToken, 100).logging())
+										.toArray(HapiSpecOperation[]::new)
+						),
+
+						burnToken(anotherToken, 1).logging(),
+						burnToken(tokenToBeQueried, 1).logging(),
+						cryptoDelete(TOKEN_TREASURY)
+								.hasKnownStatus(ACCOUNT_IS_TREASURY).logging(),
+						getTokenInfo(tokenToBeQueried).logged(),
+						getTokenInfo(anotherToken).logged(),
+						/* end token operations */
+
+						getAccountBalance(GENESIS)
+								.setNode(reconnectingNode)
+								.unavailableNode()
+				)
+				.then(
+						withLiveNode(reconnectingNode)
+								.within(5 * 60, TimeUnit.SECONDS)
+								.loggingAvailabilityEvery(30)
+								.sleepingBetweenRetriesFor(10),
+
+						/* validate tokenInfo between reconnecting node and other nodes match*/
+						getTokenInfo(tokenToBeQueried)
+								.setNode(reconnectingNode)
+								.hasAdminKey(tokenToBeQueried)
+								.hasFreezeKey(tokenToBeQueried)
+								.hasSupplyKey(tokenToBeQueried)
```

Comment:
Shouldn't the keys be `adminKey`, `freezeKey` and `supplyKey`? Could you help explain the logic here?

Reply:
HapiTokenCreate is re-saving the given keys in the registry under the `token name + a suffix` [link](https://github.com/hashgraph/hedera-services/blob/master/test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/token/HapiTokenCreate.java#L263). So, while retrieving the keys from registry we need to give tokenName such that it adds suffix and retrieveds the Key. I have an [issue](https://github.com/hashgraph/hedera-services/issues/1348) for it and will fix it.  Did not want to combine with these changes as there might be multiple tests that will change.

Reply:
Sounds good, thanks @Neeharika-Sompalli.